pikkr/parse.rs

```diff
@@ -0,0 +1,9 @@
+#![no_main]
+
+#[macro_use] extern crate libfuzzer_sys;
+extern crate pikkr;
+
+fuzz_target!(|data| {
+    let q = vec!["$.x".as_bytes()];
```

Comment:
Would it make sense to also fuzz the query parser? Either as a new target, or by splitting `data` into a query and an input part

Reply:
Possibly later. I would wait until the bugs exposed by this target have been fixed, then look at whether fuzzing the query parser discovers anything new. For now this target will tend to give easier repro cases.

Also the input JSON is much more likely to be untrusted than the query string in typical usage, so even if the query parser is buggy, most people may be comfortable using the library anyway.